include/deal.II/simplex/fe_lib.h

```diff
@@ -401,7 +401,52 @@ namespace Simplex
     get_name() const override;
   };
 
+  /**
+   * Enriched version of FE_P that can be used with nodal quadrature.
+   *
+   * Since this FE space includes bubbles (i.e., extra functions which are
+   * nonzero only on element interiors), the polynomial degrees of the component
+   * basis functions are higher than the actual approximation degree of the
+   * element. For example, with a constructor argument <code>degree = 2</code>
+   * in 3D, the polynomials are in fact cubic (degree 3) but the order of the
+   * approximation is the same as if we were using quadratic (degree 2) finite
+   * elements.
+   */
+  template <int dim, int spacedim = dim>
+  class FE_P_Bubbles : public dealii::FE_Poly<dim, spacedim>
```

Comment:
I would love to see a visualization of the shape functions! Is there a proven way to do that?

Reply:
That would be pretty easy to do by wrapping `FiniteElement::shape_value` in a `Function` and plotting it.